### PR TITLE
[Improve](point query) optimize sendFields since `writeField` is heav…

### PR DESCRIPTION
**Pull request by eldenmoon**
…y for wide tables

save about 20% FE cpu cost for point query with prepared statement which table contains 100 columns

image

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.69 seconds
 stream load tsv:          455 seconds loaded 74807831229 Bytes, about 156 MB/s
 stream load json:         25 seconds loaded 2358488459 Bytes, about 89 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230321124337_clickbench_pr_118356.html

File: fe/fe-core/src/main/java/org/apache/doris/analysis/PrepareStmt.java [hunk 1]
```diff
@@ -26,12 +26,14 @@
...
 
 import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
 import com.google.protobuf.ByteString;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.thrift.TException;
 import org.apache.thrift.TSerializer;
 
+// import java.nio.ByteBuffer;
```
Comment (xiaokang) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/PrepareStmt.java:36]: useless code

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 2]
```diff
@@ -200,8 +200,10 @@ public class StmtExecutor implements ProfileWriter {
     private boolean isCached;
     private QueryPlannerProfile plannerProfile = new QueryPlannerProfile();
     private String stmtName;
-    private PrepareStmt prepareStmt;
+    private PrepareStmt prepareStmt = null;
```
Comment (xiaokang) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:203]: java object default value is null

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 3]
```diff
@@ -1829,13 +1833,27 @@ private void sendFields(List<String> colNames, List<PrimitiveType> types) throws
...
         // send field one by one
         for (int i = 0; i < colNames.size(); ++i) {
             serializer.reset();
-            serializer.writeField(colNames.get(i), types.get(i));
-            context.getMysqlChannel().sendOnePacket(serializer.toByteBuffer());
+            if (prepareStmt != null && isExecuteStmt) {
+                // Using PreparedStatment pre serializedField to avoid serialize each time
+                // we send a field
+                byte[] serializedField = prepareStmt.getSerializedField(colNames.get(i));
+                if (serializedField != null) {
```
Comment (xiaokang) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:1845]: duplicate sendOnePacket code can be replace as follows:

```
                byte[] serializedField = prepareStmt.getSerializedField(colNames.get(i));
                if (serializedField == null) {
                    serializer.writeField(colNames.get(i), types.get(i));
                    serializedField = serializer.toArray();
                    prepareStmt.setSerializedField(colNames.get(i), serializedField);
                }

                context.getMysqlChannel().sendOnePacket(ByteBuffer.wrap(serializedField));
```

**Review by xiaokang — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/analysis/PrepareStmt.java (hunk 1 above)
Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/PrepareStmt.java]: done

**Comment by eldenmoon** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**